language/move-core/types/src/language_storage.rs

```diff
@@ -79,6 +103,153 @@ impl<'a> From<&'a TypeTag> for SafeTypeTag<'a> {
     }
 }
 
+// u8 -> depth or nesting level
+struct TypeTagVisitor(u8);
+
+use serde::de::{
+    self, DeserializeSeed, Deserializer, EnumAccess, MapAccess, SeqAccess, VariantAccess, Visitor,
+};
+
+impl<'de> Visitor<'de> for TypeTagVisitor {
+    type Value = TypeTag;
+
+    fn expecting(&self, formatter: &mut Formatter<'_>) -> Result<(), std::fmt::Error> {
+        formatter.write_fmt(format_args!("`{}`", TYPE_TAG_VARIANTS[0]))?;
+        for value in &TYPE_TAG_VARIANTS[1..] {
+            formatter.write_fmt(format_args!(", `{}`", value))?;
+        }
+        Ok(())
+    }
+
+    fn visit_enum<A>(self, data: A) -> Result<Self::Value, A::Error>
+    where
+        A: EnumAccess<'de>,
+    {
+        let (variant_type, variant) = data.variant::<DeTypeTag>()?;
+        match variant_type {
+            DeTypeTag::Bool => {
+                variant.unit_variant()?;
```

Comment:
What is this? Short comment would help the reader/reviewer the first time this appears. Not everyone knows the details of serde. I have my guesses, but...

Reply:
this one is a bit of a mess, the documentation is rather clear about this, but the lack of examples and the "step function" to arrive at how this works is certainly NOT OBVIOUS